source/particles/property_pool.cc

```diff
@@ -56,27 +96,54 @@ namespace Particles
 
 
   void
-  PropertyPool::deallocate_properties_array(Handle handle)
+  PropertyPool::deallocate_properties_array(Handle &handle)
   {
-    Assert(currently_open_handles.count(handle) == 1, ExcInternalError());
-    currently_open_handles.erase(handle);
-    delete[] handle;
+    Assert(
+      handle != invalid_handle,
+      ExcMessage(
+        "This handle is invalid and cannot be deallocated. This can happen if the "
+        "handle was deallocated already before calling this function."));
+    currently_available_handles.push_back(handle);
+    handle = invalid_handle;
+
+    // If this was the last handle, resize containers to 0.
+    // This guarantees that all future properties
+    // are allocated in a sorted way without requiring reallocation.
+    if (currently_available_handles.size() * n_properties == properties.size())
+      {
+        currently_available_handles.clear();
+        properties.clear();
+      }
   }
 
 
 
   ArrayView<double>
   PropertyPool::get_properties(const Handle handle)
   {
-    return ArrayView<double>(handle, n_properties);
+    Assert(handle != invalid_handle,
+           ExcMessage(
+             "Invalid property handle. This can happen if the "
+             "handle was deallocated before trying to access the properties."));
+
+    if (n_properties == 0)
+      return ArrayView<double>();
```

Comment:
Is this part really necessary? It seems to me that the formula below also includes this case with a modification of the formula I add below, avoiding the if statement.